src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -11,7 +19,502 @@ internal class CodeStylePage : AbstractOptionPage
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.UseImplicitTypeForIntrinsicTypes));
+            editorconfig.Append("csharp_style_var_when_type_is_apparent = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.UseImplicitTypeWhereApparent));
+            editorconfig.Append("csharp_style_var_elsewhere = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.UseImplicitTypeWherePossible));
+
+            editorconfig.AppendLine();
+            // Expression-bodied members
+            editorconfig.AppendLine("# Expression-bodied members");
+            editorconfig.Append("csharp_style_expression_bodied_methods = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedMethods));
+            editorconfig.Append("csharp_style_expression_bodied_constructors = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedConstructors));
+            editorconfig.Append("csharp_style_expression_bodied_operators = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedOperators));
+            editorconfig.Append("csharp_style_expression_bodied_properties = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedProperties));
+            editorconfig.Append("csharp_style_expression_bodied_indexers = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedIndexers));
+            editorconfig.Append("csharp_style_expression_bodied_accessors = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferExpressionBodiedAccessors));
+
+            editorconfig.AppendLine();
+            // Pattern matching preferences
+            editorconfig.AppendLine("# Pattern matching preferences");
+            editorconfig.Append("csharp_style_pattern_matching_over_is_with_cast_check = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferPatternMatchingOverIsWithCastCheck));
+            editorconfig.Append("csharp_style_pattern_matching_over_as_with_null_check = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferPatternMatchingOverAsWithNullCheck));
+
+            editorconfig.AppendLine();
+            // Null-checking preferences
+            editorconfig.AppendLine("# Null-checking preferences");
```

Comment:
we should use the same names here we use in the dialog.  i.e. 
![image](https://user-images.githubusercontent.com/4564579/42610048-701e69f4-8544-11e8-90a0-ee4d5380c0fd.png)

That way we can juse reuse the resource string we already have.

Reply:
The resource string contains extra information, i.e. Parentheses preferences is stored as "Parentheses_preferences_colon", so I was thinking it would be a bit inefficient to parse.

Reply:
yeah... i don't think the addition of hte colon is really a problem.  if this comes out a ```# Parentheses preferences:``` then that looks totally fine to me :)

Reply:
Alternatively, we could get rid of the colon.  It's not like it's serving much purpose here either:

![image](https://user-images.githubusercontent.com/4564579/42841554-4bd3c6c8-89bf-11e8-8761-9d6da24d1699.png)

--

IMO, we should match the .editorconfig emit with how we've organized things in our UI.  That means:

1. if we have more categories we want in editorconfig, then it's likely we want those in the UI as well.